pkg/controller/middleware/apikey.go

```diff
@@ -109,6 +109,7 @@ func RequireAPIKey(cacher cache.Cacher, db *database.Database, h *render.Rendere
 			// Save the authorized app on the context.
 			ctx = controller.WithAuthorizedApp(ctx, &authApp)
 			ctx = controller.WithRealm(ctx, &realm)
```

Comment:
Instead of having to remember to call `o11y.WithRealmID` after `WithRealm`, what if `WithRealm` automatically did that?